src/main/io/gps.h

```diff
@@ -86,12 +86,11 @@ typedef struct gpsCoordinateDDDMMmmmm_s {
 typedef struct gpsLocation_s {
     int32_t lat;                    // latitude * 1e+7
     int32_t lon;                    // longitude * 1e+7
-    int32_t alt;                   // altitude in 0.1m
+    int32_t alt;                    // altitude in 0.01m
```

Comment:
Is there any place where we are actually using the last digit? We always seem to be multiplying by 10 when setting. If the digit isn't used, there is no point in scaling up.

Also, if we are changing the scale of this, we need to make sure we change the scaling everywhere we are using the value.

Reply:
Agreed on the need for cleanup, and that adding a unit specifier will be helpful.
But the units should not be changed in this pull request, without a cleanup. Even if they are wrong for some places, changing them without fixing them everywhere will not gain anything.

Reply:
@mikeller ; As i understand it, this PR only changes comment and fixes MSP, actual scale change was in RTH PR ,,,

Reply:
Yep, found it: https://github.com/betaflight/betaflight/pull/5764/files#diff-62a123af09bd3b80f478b220998900f3R1023

Looking at this, the value for GPS altitude that was used in MSP was meters, not centimeters or decimeters, so we should probably use a factor of 100 in `src/main/interface/msp.c`.

Reply:
Some thoughts about altitude resolution...
Technically:
m - good for nothing; except informational displays showing absolut height
dm - good for displays; sufficient for inflight altitude adjustments; (U)INT16 will over- or underflow practically
cm - good for anything; but needs INT32 or float as data storage
I guess that was the reason why @s0up and @dbash switched to INT32 in RTH. It was needed anyway and cm resolution does not introduce a jumpy heigth input.

Configurator as one example of MSP receiver (should dm or cm send via MSP?):
Displays correct height with dm resolution if divider 10 is used in msp.c (cm -> dm). Example: "123.4 m"
With a divider of 100 in msp.c the displayed height is wrong: "12.3 m" (tested with hardware)

But i found some confusing code snippets...
msphelper.js:
    case MSPCodes.MSP_ALTITUDE:
        SENSOR_DATA.altitude = parseFloat((data.read32() / 100.0).toFixed(2)); // correct scale factor
        break; 
  Does this mean: assume mm-value, convert to dm and format to 2 decimal_places? e.g. 123456[mm] -> 123.45[m]? Read as int32, calc as float?
gps.js:
    $('.GPS_info td.alt').text((GPS_DATA.alt / 10) + ' m');
  Does this mean: assume cm-value and convert it to m with 1 decimal place?
  Note1: This would match the configurator GPS display "123.4 m".
  Note2: mikeller committed this a year ago - you should know why ;-)

This shows IMO that false and true assumptions exists in parallel. It might not be effient to generally assume a specific altitude resolution.
Sending dm's via MSP seems to fit at least the configurators code which wasn't changed.
It might be more efficient to start with the current approach, cleanup to code in the background and wait for error messages from the users.

If we would know that BF isn't used in applications with remote steering via telemetry (probably INAV's or LibrePilot's domain):
IMO it won't be life-risky to try this experimental bug finding process.
Only some telemetry displays would show irritating values...

My final proposal: 
cm resolution in INT32 for internal calculations.
dm resolution in (U)INT16 via MSP to maintain compatibility for external usage.

Reply:
@AirBreak69 :

> dm resolution in (U)INT16 via MSP to maintain compatibility for external usage.

Please read my previous comment.

There are 2 things we can do with respect to MSP:

Reply:
@mikeller 
I have to resign - although it's not my style to maintain obvious errors.
MSP API was increased to 1.39 for BF 3.4.0, RTH was injected weeks later and BF 3.40 is not final yet.
So RTH introduced a silent change but would be improved by a silent change too.

To end this strenous discussion I will scale by 0.01 in MSP translation.
But allow me to leave note about the goal of it.
Let's get this small thing out.

Sorry, one more thought: Scaling alt[cm] by 0.01 to MSP-alt[m] alone wouldn't be symmetrical. The back translation had to be adapted too. MSP-alt[m] *100 -> alt[cm]
Correct me if I'm wrong.

Reply:
Note: Configurators altitude display will be 0.1x real value after this commit.